endpoint.go

```diff
@@ -289,10 +290,29 @@ func (ep *endpoint) Join(sbox Sandbox, options ...EndpointOption) error {
 		return err
 	}
 
+	sb.Lock()
+	heap.Push(&sb.endpoints, ep)
+	sb.Unlock()
+	defer func() {
+		if err != nil {
+			var (
```

Comment:
Can we squeeze this defer block a bit.

There is no need for declaring the two vars here.
The remove from heap can be done inside the loop  and then `return`